### ISSUE TITLE
False positive for CVE-2018-3615 'Foreshadow (SGX) L1TF on i3-7100 (µcode 0xCA) with lockdown=integrity

### ISSUE THREAD

**Issue by hardfalcon** [speed47/spectre-meltdown-checker#331]
When using spectre-meltdown-checker 0.42+35+g71129d6 (which is the latest git checkout at the time of writing) on a machine with an Intel i3-7100 with the latest microcode version 0xCA, running a Archlinux x86_64 with a customized hardened Linux 5.4.2 kernel, I get a false positive for "CVE-2018-3615 'Foreshadow (SGX) L1TF" when either `lockdown=integrity` or `lockdown=confidentiality` is set on the kernel's boot cmdline. When I set `lockdown=none` instead, the false positve is gone and spectre-meltdown-checker ) claims the machine is not vulnerable (reason given: `your CPU microcode mitigates the vulnerability`).

Since the script can tell that I'm running the latest available microcode version anyhow, wouldn't it make sense to use this to also assert that CVE-2018-3615 is mitigated through this as well?

The reason for the current false positive is probably that the lockdown patch denies raw MSR access - I get an error message `Lockdown: perl: raw MSR access is restricted; see man kernel_lockdown.7` in dmesg every time I run spectre-meltdown-checker unless I boot the machine with lockdown disabled.

**Comment by speed47** [comment]
Red Hat has a similar patch but I see a different version has landed in vanilla 5.4.
I pushed a new version to master that properly detects the vanilla lockdown flavour.
Can you retry with the latest commit?

**Comment by hardfalcon** [comment]
Yes, https://github.com/speed47/spectre-meltdown-checker/commit/0cd7e1164f1ebcbcc13484fe1b1218f1154ecbb2 fixes the issue for me - thank you! :-)

I've tested it on multiple machines, and couldn't find any errors or side effect. I should add though that I've only tested it on machines that are either not affected by CVE-2018-3615 (pre-Skylake CPUs) or where it has been mitigated through microcode updates.
